test/mongodb.ts

```diff
@@ -22,6 +24,75 @@ function printExports() {
   }
 }
 
+/**
+ * Using node's require resolution logic this function will locate the entrypoint for the `'mongodb-legacy'` module.
+ * Then execute the `mongodb-legacy` module in a `vm` context that replaces the global require function with a custom
+ * implementation. The custom version of `require` will return the local instance of the driver import (magically compiled by ts-node) when
+ * the module specifier is 'mongodb' and otherwise defer to the normal require behavior to import relative files and stdlib modules.
+ * Each of the legacy module's patched classes are placed on the input object.
+ *
+ * @param exportsToOverride - An object that is an import of the MongoDB driver to be modified by this function
+ */
+function importMongoDBLegacy(exportsToOverride: Record<string, unknown>) {
+  const mongodbLegacyEntryPoint = require.resolve('mongodb-legacy');
+  const mongodbLegacyLocation = path.dirname(mongodbLegacyEntryPoint);
+  const mongodbLegacyIndex = fs.readFileSync(mongodbLegacyEntryPoint, {
+    encoding: 'utf8'
+  });
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const localMongoDB = require('../src/index');
```

Comment:
to capture the conversation from slack:
- need to add clarification/comments about how the import works and what it does so that it requires less thought to follow
- let's set up the import in a way where we can decide to turn it on or off via, say, env settings or a flag passed to a script, so that we can reuse our test suite to test new driver changes for legacy compatibility

Reply:
There's a TS doc comment on the `importMongoDBLegacy` function and it now contains all the logic so there's an env var that can control it